### PR TITLE
Infrastructure ecosystem requirements

### PR DESCRIPTION
**Pull request by mmitche**

### CODE REVIEW

**Comment by mmitche** [comment]
Any comments here or should I just merge this/

**Comment by shawnro** [comment]
I reviewed it originally and thought it looked good.  Probably worth getting a couple more eyes on it if you haven't already.

File: Documentation/InfrastructureEcosystemOverview.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+
+Specific infrastructure ecosystem requirements are implied from the set of product requirements.  The list below identifies the requirements as well as the components of the ecosystem involved in satisfying the requirement.
+- **Given an asset that is part of .NET Core, we must be able to determine the sha and repo at which that asset was produced** – After a product has shipped, it is often necessary to identify where an asset came from for the purposes of servicing or failure repro. Where possible, the sha and repo that produced the asset should be embedded within the asset.
+  - Implements Requirements
+    - Must be able to trace what has been shipped or integrated
+    - Must be able to service the product
+  - Affects Components
+    - Repository Tooling
+    - Repository Contracts
+- **Given a sha and repository that produced an asset, a functionally identical package should be producible by checking out that sha and building** - Reprodicuble builds are important, for servicing and development.
```
Comment (maririos) [inline Documentation/InfrastructureEcosystemOverview.md]: typo: Reprodicuble => Reproducible

File: Documentation/InfrastructureEcosystemOverview.md [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  - Implements Requirements
+    - Must be able to service the product
+    - Must be able to able to rapdily make changes to individual components.
```
Comment (maririos) [inline Documentation/InfrastructureEcosystemOverview.md]: typo rapdily => rapidly

File: Documentation/InfrastructureEcosystemOverview.md [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  - Affects Components
+    - Repository Tooling
+    - Repository Contracts
+- **Given a sha and repository that produced an asset, a functionally identical package should be producible by checking out that sha and building** - Reprodicuble builds are important, for servicing and development.
+  - Implements Requirements
+    - Must be able to service the product
+    - Must be able to able to rapdily make changes to individual components. 
+  - Affects Components
+    - Repository Contracts
+- **For a repository, package dependencies should be described such that the package version, the repo and sha are all specified in source** - Developers must able to locate exactly what a repository depends on.
```
Comment (maririos) [inline Documentation/InfrastructureEcosystemOverview.md]: must be able to...

File: Documentation/InfrastructureEcosystemOverview.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+- **Build/Test System** – The mechanics of building/testing a repository within a single host environment.  The build system is not concerned with environment procurement, host CI systems, etc.
+- **CI/Build Automation** – The set of software and services that automates the procurement of environments, scheduling of CI jobs, feedback to GitHub, etc.  Generally, this is VSTS with a set of additional functionality provided by .NET Core.
+- **Repository Tooling** – A set of standalone tooling that is used to interact with the repository.  Examples include tooling to read repository package information, tooling to update repository package info, etc.
+- **Repository API** – A set of primitive commands common to each repository used to perform typical repository actions.
+- **Dependency Flow Automation** – A set of tools and services used to automate the movement of dependencies among repositories based on configurable policies. Maestro# implements most of this functionality, utilizing the repository tooling.
```
Comment (maririos) [inline Documentation/InfrastructureEcosystemOverview.md]: Maestro++

File: Documentation/InfrastructureEcosystemOverview.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+- **Visualization/Reporting System** – A set of services used to report the status of the product.  Currently implemented in Mission Control
```
Comment (maririos) [inline Documentation/InfrastructureEcosystemOverview.md]: Should we include the link to MC?

**Pull request merged**